src/components/Paginate.tsx

```diff
@@ -0,0 +1,63 @@
+import clsx from 'clsx';
+import React, { useState } from 'react';
+
+interface PaginationProps {
+  currentPage: number
+  setPage: (page: number) => void;
+  totalCount: number;
+  perPage: number;
+}
+
+const Pagination: React.FC<PaginationProps> = ({ currentPage, setPage, totalCount, perPage }) => {
+  const totalPages = Math.ceil(totalCount / perPage);
+  const pageNumbersToShow = 5;
+
+  const generatePageNumbers = () => {
+    const pageNumbers = [];
+    const startPage = Math.max(1, currentPage - Math.floor(pageNumbersToShow / 2));
+
+    for (let i = startPage; i <= Math.min(totalPages, startPage + pageNumbersToShow - 1); i++) {
+      pageNumbers.push(i);
+    }
+
+    return pageNumbers;
+  };
+
+  const handlePageClick = (page: number) => {
+    setPage(page);
+  };
```

Comment:
not sure why we create a separate function here, we could've used `setPage` itself everywhere. 🤷